### PR TITLE
fix(cemu): add ltcg, build with gcc9Stdenv

### PR DESCRIPTION
**Pull request by mattchrist**
###### Motivation for this change
Fixes CEmu.

ZHF: #144627

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mattchrist** [comment]
@luc65r I should have searched first, I now see you tried to fix this previously in #114577

I expect this will have the same MacOS issues you encountered. Should this package get marked as broken on darwin?

**Comment by luc65r** [comment]
> Should this package get marked as broken on darwin?

If you can't test on darwin, I think so.

File: pkgs/applications/science/math/cemu/ltcg.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/gui/qt/CEmu.pro b/gui/qt/CEmu.pro
+index 1e509d12..107df284 100644
+--- a/gui/qt/CEmu.pro
++++ b/gui/qt/CEmu.pro
+@@ -52,7 +52,7 @@ TEMPLATE = app
+ # Localization
+ TRANSLATIONS += i18n/fr_FR.ts i18n/es_ES.ts i18n/nl_NL.ts
+ 
+-CONFIG += c++11 console
++CONFIG += c++11 console ltcg
```
Comment (luc65r) [inline pkgs/applications/science/math/cemu/ltcg.patch:10]: This patch could be avoided by adding `"CONFIG+=ltcg"` in `qmakeFlags`

**Review by luc65r — APPROVED** [review]

File: pkgs/applications/science/math/cemu/ltcg.patch (hunk 1 above)
Comment (mattchrist) [reply pkgs/applications/science/math/cemu/ltcg.patch]: Neat! Thanks

**Review by luc65r — APPROVED** [review]

**Comment by thiagokokada** [comment]
Please fix ec09e96 to follow https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes.


```
cemu: add link-time-compile-gen, stdexcept 
```

**Comment by mattchrist** [comment]
@thiagokokada Fixed, thanks!

**Comment by thiagokokada** [comment]
Could you also squash 7493baa and fefdd46? Because it seems fefdd46 just do part what is done on 7493baa, just in another way.

File: pkgs/applications/science/math/cemu/default.nix [hunk 2]
```diff
@@ -38,8 +39,13 @@ mkDerivation rec {
     libpng_apng
   ];
 
+  patches = [
+    ./stdexcept.patch
```
Comment (jonringer) [inline pkgs/applications/science/math/cemu/default.nix:43]: nothing in this patch looks nixos specific. is there a way to upstream it?

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 144889` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cemu</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/applications/science/math/cemu/default.nix (hunk 2 above)
Comment (mattchrist) [reply pkgs/applications/science/math/cemu/default.nix]: I just pushed changes that build without patching, by using gcc9Stdenv.

Comment (mattchrist) [reply pkgs/applications/science/math/cemu/default.nix]: I'm sure upstream could be fixed to build with stdenv. I'm not personally interested in using this package or contributing upstream for this package and it's dependencies (the patch touched submodules too).
My interest is in getting it to build again for ZHF.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 144889 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>cemu</li>  </ul></details>

**Pull request merged**